doc/design/determinism_annotation.md

```diff
@@ -0,0 +1,58 @@
+# Determinism Annotations
+
+## Cert
+
+All functions are to be certified, hence there will be no label to indicate this.
```

Comment:
You should mention here that this applies only to the CMake package with [quality level 1+](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#quality-levels).